homeassistant/util/unit_conversion.py

```diff
@@ -138,6 +144,15 @@ def get_unit_ratio(cls, from_unit: str | None, to_unit: str | None) -> float:
         from_ratio, to_ratio = cls._get_from_to_ratio(from_unit, to_unit)
         return from_ratio / to_ratio
 
+    @classmethod
+    @lru_cache
+    def _are_unit_inverses(cls, from_unit: str | None, to_unit: str | None) -> bool:
+        """Return true if units are inverses to one another."""
+        return (from_unit, to_unit) in cls._UNIT_INVERSES or (
+            to_unit,
+            from_unit,
+        ) in cls._UNIT_INVERSES
```

Comment:
I still think this is a bigger maintenance burden, and you are more likely to miss adding "pairs" as you add more units.

```suggestion
    def _are_unit_inverses(cls, from_unit: str | None, to_unit: str | None) -> bool:
        """Return true if one unit is an inverse but not the other."""
        return (from_unit in cls._UNIT_INVERSES) != (to_unit in cls._UNIT_INVERSES)
```

Reply:
Fixed in c620c2317735ecc53301ea18862264a9fdde8aae.

Reply:
See maintenance burden: using tuples means `(n_regular * n_inverted)` whereas using a simple set means `(1 * n_inverted)`

# Sample with three regular and two inverted:
```python
    _UNIT_INVERSES: set[tuple[str, str]] = {
        (U_1, IU_1),
        (U_2, IU_1),
        (U_3, IU_1),
        (U_1, IU_2),
        (U_2, IU_2),
        (U_3, IU_2),
    }

# vs
    _UNIT_INVERSES: set[str] = {
        IU_1,
        IU_2,
    }
```

Reply:
I agree this is a more elegant solution.